### PR TITLE
Replace flake with flakeheaven for linting

### PR DESCRIPTION
**Pull request by matteopilz**
The pyproject.toml needs to be configured. We can also optionally exclude not linted file types automatically and end the action early https://github.com/tj-actions/changed-files.

### CODE REVIEW

**Comment by matteopilz** [comment]
I have added the plugins and tested it on push. It still needs to be configured and I don't know which settings we want.

**Comment by jpfeuffer** [comment]
Any settings are fine for me.
@OpenMS/core What do you think about auto-applying linting changes with e.g., https://github.com/adamchainz/blacken-docs

**Comment by matteopilz** [comment]
Does blacken-docs work only for code blocks?

**Comment by jpfeuffer** [comment]
Yes, what else do you want to lint?

**Comment by matteopilz** [comment]
The markdown in the rst files, basically apply the suggested changes for errors or warnings from flakeheaven automatically. I guess this would be too insecure.

**Comment by matteopilz** [comment]
Do we then actually still need flakeheaven, if we correct code blocks anyway?

**Comment by jpfeuffer** [comment]
If we agree to auto-apply, no, sorry. I was actually assuming that flake heaven can auto-apply.

**Comment by matteopilz** [comment]
blacken needs pre-commit (https://pre-commit.com/#install) installed on the machine to register a commit and actually change something beforehand. How should we handle this? If we install it with a commit on a runner, it kind of defeats the purpose because nothing will be changed.

**Comment by jpfeuffer** [comment]
you can also just run it in an action. Pre-commit is just the preferred way for most users, since they want to leave the choice to developers I guess.

**Comment by jpfeuffer** [comment]
`python -m pip install blacken-docs && blacken-docs -l 80 docs/source/*.rst`

**Comment by jpfeuffer** [comment]
That's how I did #333

**Comment by jpfeuffer** [comment]
By the way, I changed my mind for line length as you can see. BEcause this is for documentation. And it should be visible on mobile as well.
No one is developing in there.

**Comment by jpfeuffer** [comment]
You need to somehow suggest or even directly push the changes.
I guess the former (suggesting) is a nice middle ground. And for follow up PRs probably not to huge.

https://github.com/reviewdog/action-suggester

**Comment by jpfeuffer** [comment]
If you only use pull_request as a trigger, which is fine in this case actually, you will not be able to test it. Unless someone does a PR against YOUR branch.
(your action is not on master, against which this PR is made)

**Comment by jpfeuffer** [comment]
Do a  PR with the yml only and I will merge it.

**Comment by matteopilz** [comment]
I think action-suggester works only on pull_request. I'm currently not sure, if it will work, but we can try it. I'm removing the other files.

File: .github/workflows/code-blocks-linting.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        pip install blacken-docs
+
+    - name: Get changed files
+      id: changed-files
+      uses: tj-actions/changed-files@v35
+
+    - name: Run action-suggester
+      uses: reviewdog/action-suggester@v1
+      with:
+        tool_name: blacken-docs -l 80 ${{ steps.changed-files.outputs.all_changed_files }}
```
Comment (jpfeuffer) [inline .github/workflows/code-blocks-linting.yaml:24]: No, I think toolname is just for display. It was correct before.

Comment (jpfeuffer) [reply .github/workflows/code-blocks-linting.yaml:24]: i.e. with a separate step that actually runs blacken-docs

**Comment by matteopilz** [comment]
It didn't work, blacken still exits with an error.

**Comment by matteopilz** [comment]
It looks like flakeheaven would actually report in the output format, action-suggester expects...

**Comment by jpfeuffer** [comment]
I doubt it, since it is not able to make any changes. You might be able to comment with the comments. But this is almost useless. You have to go back to the code, search the place where the comments were etc.
I just want to click on accept suggestion and be done with it.

**Comment by jpfeuffer** [comment]
Also: flakeheaven is also bugged: https://github.com/flakeheaven/flakeheaven/issues/165

**Pull request merged**